### PR TITLE
#12625: Add Current Date to update_time Field for Block and Pages

### PR DESCRIPTION
**Pull request by osrecio**
Set Current Date Time to update_time field

### Description
add current date time to 'update_time' in `_beforeSave` method  for Page and Block

### Fixed Issues (if relevant)
1. magento/magento2#12625: when saving a page in magento 2.2.1, 'Modified' date field is not getting updated

### Manual testing scenarios
1. Login to magento admin
2. Go to Content > Pages
3. In Action column, click select > edit
4. Make changes to that page and click 'Save Page'

### Contribution checklist
 - [X] Pull request has a meaningful description of its purpose
 - [X] All commits are accompanied by meaningful commit messages
 - [X] All new or changed code is covered with unit/integration tests (if applicable)
 - [X] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by mzeis** [comment]
Hi @osrecio,

thanks for your pull request! Please can you have a look at the failing integration test?

**Comment by osrecio** [comment]
@mzeis Adapted Tests

File: app/code/Magento/Cms/Model/ResourceModel/Block.php [hunk 1]
```diff
@@ -91,6 +92,10 @@ protected function _beforeSave(AbstractModel $object)
                 __('A block identifier with the same properties already exists in the selected store.')
             );
         }
+
+        $date = (new \DateTime())->format(DateTime::DATETIME_PHP_FORMAT);
```
Comment (sidolov) [inline app/code/Magento/Cms/Model/ResourceModel/Block.php]: Please, use framework implementation for date value \Magento\Framework\Stdlib\DateTime\DateTime::date

**Review by sidolov — CHANGES_REQUESTED** [review]

**Comment by osrecio** [comment]
@sidolov Changes made, and added integration tests for Cms Block 😃

File: dev/tests/integration/testsuite/Magento/Cms/Model/BlockTest.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        );
+        $session->setUser($user);
+    }
+
+    /**
+     * @magentoDbIsolation enabled
+     */
+    public function testUpdateTime()
+    {
+        $objectManager = Bootstrap::getObjectManager();
```
Comment (mzeis) [inline dev/tests/integration/testsuite/Magento/Cms/Model/BlockTest.php]: It's better to get the object manager in the `setUp` method ([example](https://github.com/magento/magento2/blob/3e3e8fc90ace43ef7113e52f2c1e7bc20c78d94a/dev/tests/integration/testsuite/Magento/Catalog/Model/Product/CopierTest.php#L67)).

**Review by mzeis — CHANGES_REQUESTED** [review]

File: dev/tests/integration/testsuite/Magento/Cms/Model/BlockTest.php (hunk 2 above)
Comment (osrecio) [reply dev/tests/integration/testsuite/Magento/Cms/Model/BlockTest.php]: Added ObjectManager to Setup.

I copied test from `dev/tests/integration/testsuite/Magento/Cms/Model/PageTest.php`

**Comment by osrecio** [comment]
@mzeis Added ObjectManager to `setUp`.

I copied test from `dev/tests/integration/testsuite/Magento/Cms/Model/PageTest.php`

**Comment by osrecio** [comment]
@sidolov I followed your [suggestions](https://github.com/magento/magento2/pull/12637#issuecomment-352423587) and I adapted my PR.

**Review by mzeis — APPROVED** [review]

**Comment by mzeis** [comment]
@osrecio When comparing the changes with #12625, I noticed that the implementation changes are the same but the tests are carried out differently. Would it make sense to adjust the integration tests here to make them equal to `2.3-develop`?

**Comment by osrecio** [comment]
@mzeis Are you referring to the tests changed in branch `2.3-develop` of my other [PR](https://github.com/magento/magento2/pull/12637)?

These test: [BlockTest.php](https://github.com/magento/magento2/pull/12637/files#diff-5bea032a48a5d9d5f7b04ab79ccdef61) , [PageTest.php](https://github.com/magento/magento2/pull/12637/files#diff-9652dfbcb5f5efadf88bac53fa585dd0)

Thanks!

**Comment by mzeis** [comment]
@osrecio Yes exactly. Since Stanislav approved your changes in #12637 I thought it might be a good idea to keep the code between 2.2 and 2.3 as similar as possible.

**Comment by osrecio** [comment]
I will work to make these changes, Thanks @mzeis

**Comment by osrecio** [comment]
Hi @mzeis I adapted `BlockTest.php` like `2.3-develop` in `PageTest.php` nothing to do because other methods in tests covering new funcionality not included in `2.2-develop`.

**Review by sidolov — APPROVED** [review]

**Comment by sidolov** [comment]
@osrecio, please, check failed static tests

**Comment by osrecio** [comment]
@sidolov Fixed Static Tests, the [pre-commit](https://github.com/magento/magento2/blob/2.2-develop/dev/tools/Magento/Tools/StaticReview/pre-commit) deleted by me the reference in namespace 😄

**Comment by mzeis** [comment]
@osrecio Many thanks for your contribution. It has been merged. Happy holidays! 🎄

**Comment by osrecio** [comment]
Thanks @mzeis , Happy holidays and happy new year!!

**Pull request merged**